File: packages/java/parser-jvm-plugin-backbone/src/test/java/com/vaadin/hilla/parser/plugins/backbone/multiendpoints/MultiEndpointsFooEndpoint.java
package com.vaadin.hilla.parser.plugins.backbone.multiendpoints;

import com.vaadin.hilla.Endpoint;

@Endpoint
public class MultiEndpointsFooEndpoint {
    public String getFoo() {
        return "foo";
    }

    public MultiEndpointsSharedModel getShared() {
        return new MultiEndpointsSharedModel();
    }
}

File: packages/java/typescript-generator/src/test/resources/com/vaadin/hilla/parser/plugins/backbone/multiendpoints/snapshots/MultiEndpointsFooEndpoint.ts
import { EndpointRequestInit } from "@vaadin/hilla-frontend";
import type MultiEndpointsSharedModel from "./com/vaadin/hilla/parser/plugins/backbone/multiendpoints/MultiEndpointsSharedModel.js";
import client from "./connect-client.default.js";
async function getFoo(init?: EndpointRequestInit): Promise<string | undefined> { return client.call("MultiEndpointsFooEndpoint", "getFoo", {}, init); }
async function getShared(init?: EndpointRequestInit): Promise<MultiEndpointsSharedModel | undefined> { return client.call("MultiEndpointsFooEndpoint", "getShared", {}, init); }
export { getFoo, getShared };
